Graphics/GraphicsEngineWebGPU/include/BufferWebGPUImpl.hpp

```diff
@@ -46,6 +47,15 @@ class BufferWebGPUImpl final : public BufferBase<EngineWebGPUImplTraits>
 public:
     using TBufferBase = BufferBase<EngineWebGPUImplTraits>;
 
+    struct StagingBufferSyncInfo
```

Comment:
⚠️ This data is passed to the callback. It needs to be derived from IObject and AddRef'ed before it is passed to the callback.

Reply:
No need to complicate the code; the object lifetimes are well-defined. I doubt anyone would call 16 flushes inside a frame